cli/src/main/java/org/infinispan/cli/commands/CLI.java

```diff
@@ -233,6 +230,35 @@ public CommandResult exec(ContextAwareCommandInvocation invocation) {
       }
    }
 
+   public static void configureSslContext(Context context, Resource truststore, String truststorePassword, Resource keystore, String keystorePassword, String hostnameVerifier, boolean trustAll) throws UnrecoverableKeyException, KeyStoreException, NoSuchAlgorithmException, IOException {
+      String sslTrustStore = truststore != null ? truststore.getAbsolutePath() : context.getProperty(Context.Property.TRUSTSTORE);
+      String sslKeyStore = keystore != null ? keystore.getAbsolutePath() : context.getProperty(Context.Property.KEYSTORE);
+      if (sslTrustStore != null) {
+         TrustManagerFactory trustManagerFactory;
+         String sslTrustStorePassword = truststorePassword != null ? truststorePassword : context.getProperty(Context.Property.TRUSTSTORE_PASSWORD);
+         try (FileInputStream f = new FileInputStream(sslTrustStore)) {
+            KeyStore ts = KeyStoreUtil.loadKeyStore(ProviderUtil.INSTALLED_PROVIDERS, null, f, sslTrustStore, sslTrustStorePassword != null ? sslTrustStorePassword.toCharArray() : null);
+            trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(ts);
+         }
+         HostnameVerifier verifier = hostnameVerifier != null ? new RegexHostnameVerifier(hostnameVerifier) : null;
+         KeyManagerFactory keyManagerFactory = null;
+         if (sslKeyStore != null) {
```

Comment:
This won't work if the user has configured `--keystore --trustall` as the client certificate won't be passed to the server as no truststore exists.

Reply:
Well spotted, I've reworked the logic